### PR TITLE
ensure we drop into the right continuation cb

### LINKED ISSUE #4408: Setting context to WEBVIEW upon autoWebview = true causes the ios_remote_debugger to wait indefinitely
I am experiencing a very frustrating experience on iOS trying to run my tests and not being able to. The problem I am facing now occurs when I try to create the Appium Driver through dot-net-driver in my C# tests.

- My Appium server runs on a Mac machine: OSX 10.9.5.
- Appium 1.3.4
- Running against iOS device OS version 7.1.1
- XCode version 6.1.1

Tests asks for the following capabilities:

    DesiredCapabilities capabilities = new DesiredCapabilities();
    capabilities.SetCapability("appium-version", "1.0");
    capabilities.SetCapability("browserName", "");
    capabilities.SetCapability("autoWebview", true);
    capabilities.SetCapability("app", "/Users/myuser/IPAs/myipa.ipa");
    capabilities.SetCapability("platformName", "iOS");
    capabilities.SetCapability("platformVersion", "7.1");
    capabilities.SetCapability("deviceName", "iOS");
    capabilities.SetCapability("udid", "bf74855...");

When I start the test the app is already installed in the iPad. Also the iPad is correctly connected through `ios_webkit_debug_proxy` whose server is correctly running on port 27753.

##Appium log
This is the log when the test reaches:

    AppiumDriver driver = new AppiumDriver("<server addr>", capabilities, TimeSpan.FromSeconds(180));

> info: Welcome to Appium v1.3.4 (REV c8c79a85fbd6870cd6fc3d66d038a115ebe22efe)
info: Appium REST http interface listener started on 0.0.0.0:4723
info: Console LogLevel: debug
info: --> POST /wd/hub/session {"desiredCapabilities":{"appium-version":"1.0","browserName":"","autoWebview":true,"app":"/Users/navdevuser/APKs/crete_msapps.ipa","platformName":"iOS","platformVersion":"7.1","deviceName":"iOS","udid":"93332e84baf4b72b895bf7892d37aca5880b0dba"}}
info: Client User-Agent string: undefined
info: [debug] The following desired capabilities were provided, but not recognized by appium. They will be passed on to any other services running on this server. : appium-version
info: [debug] Using local .ipa from desired caps: /Users/navdevuser/APKs/crete_msapps.ipa
info: [debug] Copying local zip to tmp dir
info: [debug] /Users/navdevuser/APKs/crete_msapps.ipa copied to /var/folders/4r/1_9mg9550zvchwlw96v0033c0000gr/T/115019-7728-gra27v/appium-app.zip
info: [debug] Unzipping /var/folders/4r/1_9mg9550zvchwlw96v0033c0000gr/T/115019-7728-gra27v/appium-app.zip
info: [debug] Testing zip archive: /var/folders/4r/1_9mg9550zvchwlw96v0033c0000gr/T/115019-7728-gra27v/appium-app.zip
info: [debug] Zip archive tested clean
info: [debug] Unzip successful
info: [debug] Using locally extracted app: /var/folders/4r/1_9mg9550zvchwlw96v0033c0000gr/T/115019-7728-gra27v/Payload/DynamicsNAV.app
info: [debug] Creating new appium session 7b22176f-3446-4372-a268-1cf208f5cacd
info: [debug] Removing any remaining instruments sockets
info: [debug] Cleaned up instruments socket /tmp/instruments_sock
info: [debug] Setting Xcode folder
info: [debug] Setting Xcode version
info: [debug] Setting iOS SDK Version
info: [debug] Getting sdk version from xcrun with a timeout
info: [debug] iOS SDK Version set to 8.1
info: [debug] Not checking whether simulator is available since we're on a real device
info: [debug] Detecting automation tracetemplate
info: [debug] Not auto-detecting udid, running on sim
info: [debug] Parsed app Info.plist (as binary)
info: [debug] Parsed app Localizable.strings
info: [debug] Getting bundle ID from app
info: [debug] Parsed app Info.plist (as binary)
info: [debug] Creating instruments
info: On some xcode 6 platforms, instruments-without-delay does not work. If you experience this, you will need to re-run appium with the --native-instruments-lib flag
info: [debug] Preparing uiauto bootstrap
info: [debug] Dynamic bootstrap dir: /Users/navdevuser/Library/Application Support/appium/bootstrap
info: [debug] Dynamic env: {"nodePath":"/usr/local/Cellar/node/0.10.31/bin/node","commandProxyClientPath":"/usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js","instrumentsSock":"/tmp/instruments_sock","interKeyDelay":null,"justLoopInfinitely":false,"autoAcceptAlerts":false,"sendKeyStrategy":"grouped"}
...o: [debug] Dynamic bootstrap code: // This file is automatically generated. Do not manually modify!
info: [debug] Dynamic bootstrap path: /Users/navdevuser/Library/Application Support/appium/bootstrap/bootstrap-555da0501182b627.js
info: [debug] Reusing dynamic bootstrap: /Users/navdevuser/Library/Application Support/appium/bootstrap/bootstrap-555da0501182b627.js
info: [debug] Getting device string from opts: {"forceIphone":false,"forceIpad":false,"xcodeVersion":"6.1.1","iOSSDKVersion":"8.1","deviceName":"iOS","platformVersion":"7.1"}
info: [debug] fixDevice is on
info: [debug] Final device string is: 'iOS (7.1 Simulator)'
info: [debug] Not setting device type since we're on a real device
info: [debug] Checking whether we need to set app preferences
info: [debug] Not setting iOS and app preferences since we're on a real device
info: [debug] Running ios sim reset flow
info: [debug] Killing the simulator process
info: [debug] Killall iOS Simulator
info: [debug] Killing any other simulator daemons
info: [debug] On a real device; cannot clean device state
info: [debug] Not setting locale because we're using a real device
info: [debug] No iOS / app preferences to set
info: [debug] Starting iOS device log capture via deviceconsole
info: [debug] Not pre-launching simulator
info: [debug] Creating iDevice object with udid 93332e84baf4b72b895bf7892d37aca5880b0dba
info: [debug] Couldn't find ideviceinstaller, trying built-in at /usr/local/lib/node_modules/appium/build/libimobiledevice-macosx/ideviceinstaller
info: [debug] Checking app install status using: /usr/local/lib/node_modules/appium/build/fruitstrap/fruitstrap isInstalled --id 93332e84baf4b72b895bf7892d37aca5880b0dba --bundle com.microsoft.dynamics.nav
info: [debug] App is installed.
info: [debug] fullReset not requested. No need to install.
info: [debug] Starting command proxy.
info: [debug] Instruments socket server started at /tmp/instruments_sock
info: [debug] Starting instruments
info: [debug] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments
info: Launching instruments
info: [debug] Attempting to run app on real device with UDID 93332e84baf4b72b895bf7892d37aca5880b0dba
info: [debug] Spawning instruments with command: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments -t /Applications/Xcode.app/Contents/Applications/Instruments.app/Contents/PlugIns/AutomationInstrument.xrplugin/Contents/Resources/Automation.tracetemplate -D /tmp/appium-instruments/instrumentscli0.trace -w 93332e84baf4b72b895bf7892d37aca5880b0dba com.microsoft.dynamics.nav -e UIASCRIPT "/Users/navdevuser/Library/Application Support/appium/bootstrap/bootstrap-555da0501182b627.js" -e UIARESULTSPATH /tmp/appium-instruments
info: [debug] And extra without-delay env: {}
info: [debug] And launch timeouts (in ms): {"global":90000}
info: [debug] [INST STDERR] 2015-01-19 16:39:28.142 instruments[7755:1003] WebKit Threading Violation - initial use of WebKit from a secondary thread.
info: Instruments is ready to receive commands
info: [debug] Instruments launched. Starting poll loop for new commands.
info: [debug] Setting bootstrap config keys/values
info: [debug] Pushing command to appium work queue: "target = $.target();\nau = $;\n$.isVerbose = true;\n"
info: [debug] Socket data received (2 bytes)
info: [debug] Socket data being routed.
info: [debug] Sending command to instruments: target = $.target();
au = $;
$.isVerbose = true;
info: [debug] [INST] 2015-01-19 15:39:33 +0000 Debug: evaluation finished
info: [debug] Socket data received (27 bytes)
info: [debug] Socket data being routed.
info: [debug] Got result from instruments: {"status":0,"value":true}
info: [debug] Setting initial orientation to PORTRAIT
info: [debug] Pushing command to appium work queue: "au.setScreenOrientation('PORTRAIT')"
info: [debug] Sending command to instruments: au.setScreenOrientation('PORTRAIT')
info: [debug] [INST] 2015-01-19 15:39:33 +0000 Debug: responding with:
info: [debug] [INST] 2015-01-19 15:39:33 +0000 Debug: Running system command #1: /usr/local/Cellar/node/0.10.31/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /tmp/instruments_sock 2,{"status":0,"value":true}...
info: [debug] [INST] 2015-01-19 15:39:34 +0000 Debug: Got new command 1 from instruments: au.setScreenOrientation('PORTRAIT')
info: [debug] [INST] 2015-01-19 15:39:34 +0000 Debug: evaluating au.setScreenOrientation('PORTRAIT')
info: [debug] [INST] 2015-01-19 15:39:34 +0000 Debug: target.setDeviceOrientation("1")
info: [debug] [INST] 2015-01-19 15:39:35 +0000 Debug: evaluation finished
info: [debug] Socket data received (33 bytes)
info: [debug] Socket data being routed.
info: [debug] Got result from instruments: {"status":0,"value":"PORTRAIT"}
info: [debug] Setting auto webview
info: [debug] [INST] 2015-01-19 15:39:35 +0000 Debug: responding with:
info: [debug] [INST] 2015-01-19 15:39:35 +0000 Debug: Running system command #2: /usr/local/Cellar/node/0.10.31/bin/node /usr/local/lib/node_modules/appium/node_modules/appium-uiauto/bin/command-proxy-client.js /tmp/instruments_sock 2,{"status":0,"value":"PORTRAIT"}...
info: [debug] Navigating to most recently opened webview
info: [debug] [REMOTE] Getting WebKitRemoteDebugger pageArray
info: [debug] Picking webview WEBVIEW_1
info: [debug] Attempting to set context to 'WEBVIEW_1'
info: [debug] [REMOTE] Debugger web socket connected to url [ws://localhost:27753/devtools/page/1]
info: [debug] [REMOTE] Unregistering from page readiness notifications
info: [debug] Waiting for app source to contain elements
info: [debug] Pushing command to appium work queue: "au.mainApp().getTreeForXML()"
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done
info: [debug] We're in the middle of processing a remote debugger command, waiting to run next command until done

I have only one webview and when that webview is set, this problem occurs, the message `We're in the middle of processing a remote debugger command, waiting to run next command until done` is displayed forever.

What's happening? Thank you

### PR DESCRIPTION
**Pull request by jlipps**
...when selecting hybrid context in ios (fix #4408)

### CODE REVIEW

**Comment by jlipps** [comment]
cc @sebv

File: lib/devices/ios/ios-controller.js [hunk 1]
```diff
@@ -1618,6 +1619,7 @@ iOSController.setContext = function (name, callback, skipReadyCheck) {
               this.remote.disconnect(function () {
                 this.curContext = idx;
                 this.remote.connect(idx, function () {
+                  this.processingRemoteCmd = false;
```
Comment (sebv) [inline lib/devices/ios/ios-controller.js]: missing `bind`

**Comment by sebv** [comment]
except that lgtm, maybe we should avoid calling callbacks method `onDone`.

File: lib/devices/ios/ios-controller.js (hunk 1 above)
Comment (jlipps) [inline lib/devices/ios/ios-controller.js]: fixed, thx

**Comment by jlipps** [comment]
any reason not to merge this?

**Pull request merged**